### DISCUSSION TITLE
Chapter_1

### DISCUSSION THREAD

**Discussion by hongsam14** [Kookmin-Univ-C-programming/SnakeGame#10] [Ideas]
![chapter_1](https://user-images.githubusercontent.com/69339846/168425691-7aa1e889-af7a-4fb8-92bf-98058239f15d.JPG)

**Comment by hongsam14** [comment]
단순히 생각하면 맵 데이터를 저장하는 공간이 있어야 할 것 같고 게임을 시작할 때 맵데이터를 읽겠네요. 2차원 배열이면 충분할까요?

**Reply by mumat0103** [reply]
2차원 배열이면 충분할 것 같아요

**Comment by mumat0103** [comment]
맵 만드는 함수를 정의해서 쓰는 게 어떨까요? void initialmap(int row, int col) 이런 식으로 받아서 처리하면 괜찮을 것 같아요

**Reply by hongsam14** [reply]
함수여도 될것같고 클래스여도 될것 같네요. 클래스인데 생성자로 row col 받는 식으로

**Reply by hongsam14** [reply]
#11 맵 만드는 클래스 짜보았습니다.

**Comment by hongsam14** [comment]
GameField Class에 합치는데 성공하였고, 누수는 없는 것 같습니다. initialmap.h 와 map.h를 cpp파일로 분리하는 작업은 어떡하실 건가요? 사실 이미 주요 기능들을 GameField에 합쳐서 코드 상에서 접근할 일은 거의 없을 것 같긴 합니다.

**Reply by mumat0103** [reply]
initialmap.h 와 map,h 파일에서 필요한 부분을 GameField에 합쳤기 때문에, 분리하려면 할 수는 있을 것 같은데 굳이 이 파일들을 남겨놔야할까 라는 생각이 듭니다. 어떻게 생각하세요??

**Reply by hongsam14** [reply]
음 맞는말 같습니다

그럼 지워서 올릴께요

**Reply by hongsam14** [reply]
바로 master에 푸쉬하겠습니다.

**Reply by mumat0103** [reply]
네 알겠습니다.

**Reply by hongsam14** [reply]
push 하였고 commit 메시지로 간단하게 정리하였습니다. 나중에 서류 정리할 때 issue, commit, pull request, discussion 이렇게 기록 다 남겨놓으니 참고하면 될것 같아요.

**Reply by mumat0103** [reply]
넵 확인했습니다
